### PR TITLE
Upload videos in administration

### PR DESCRIPTION
**Pull request by illusionOfParadise**

### CODE REVIEW

File: src/Controller/AssetsController.php [hunk 1]
```diff
@@ -32,6 +32,12 @@ public function assetAction(string $path, Request $request)
     {
         $parameters = $request->query->all();
 
+        if (array_key_exists('is_video', $parameters) && $parameters['is_video']) {
+            return new Response($this->get('app.storage')->read($path), Response::HTTP_OK, [
+                'Content-Type' => $parameters['mime_type'],
```
Comment (salahm) [inline src/Controller/AssetsController.php]: `ContentType`

File: src/Controller/AssetsController.php [hunk 2]
```diff
@@ -32,6 +32,12 @@ public function assetAction(string $path, Request $request)
     {
         $parameters = $request->query->all();
 
+        if (array_key_exists('is_video', $parameters) && $parameters['is_video']) {
```
Comment (salahm) [inline src/Controller/AssetsController.php]: `if (!empty($parameters['is_video']))`

File: src/Entity/Media.php [hunk 3]
```diff
@@ -281,12 +283,24 @@ public function setFile(File $file = null): Media
             return $this;
         }
 
+        if ($this->isVideo()) {
+            $this->width = 0;
+            $this->height = 0;
+            $this->mimeType = $file->getMimeType();
+        } else {
+            $this->width = $infos[0];
```
Comment (salahm) [inline src/Entity/Media.php]: please inline : `$this->width = $this->isVideo() ?  0 : $infos[0]`

File: src/Entity/Media.php [hunk 4]
```diff
@@ -281,12 +283,24 @@ public function setFile(File $file = null): Media
...
-        $this->width = $infos[0];
-        $this->height = $infos[1];
-        $this->mimeType = $infos['mime'];
 
         return $this;
     }
+
+    public function isVideo(): bool
+    {
+        return strpos($this->mimeType, 'video') > -1;
```
Comment (salahm) [inline src/Entity/Media.php]: `return false !== strpos($this->mimeType, 'video')`

File: src/Entity/Media.php (hunk 3 above)
Comment (HeahDude) [inline src/Entity/Media.php]: I wouldn't do that, it would add unneeded extra calls.

File: src/Entity/Media.php [hunk 5]
```diff
@@ -281,12 +283,24 @@ public function setFile(File $file = null): Media
             return $this;
         }
 
+        if ($this->isVideo()) {
```
Comment (HeahDude) [inline src/Entity/Media.php]: This check should be done on the new value being set.

Comment (HeahDude) [inline src/Entity/Media.php]: This [line](https://github.com/EnMarche/en-marche.fr/pull/1404/files#diff-68a5398926b02f9663cde4e3d436fe02R282) may be a problem to get here

File: templates/admin/media/list_thumbnail.html.twig [hunk 6]
```diff
@@ -2,6 +2,10 @@
 
 {% block field %}
     <div style="text-align: center;">
-        <img src="{{ media_asset(object, {'w': 150}) }}" style="max-height: 50px; max-width: 100px;" />
+        {% if object.isVideo() %}
```
Comment (HeahDude) [inline templates/admin/media/list_thumbnail.html.twig]: `object.isVideo()` => `object.video`

File: templates/home/articles.html.twig [hunk 7]
```diff
@@ -1,14 +1,21 @@
...
         <div class="csv card--small card--small__video {{ block.hasDisplayFilter ? '' : ' card--no-gradient' }}">
-            <a href="{{ block.link }}" class="background-container">
-                <div class="background card__background progressive-background"
-                     data-sd="{{ media_asset(block.media, {'w': 300, 'q': 20}) }}"
-                     data-hd="{{ media_asset(block.media, {'w': 1200, 'q': 70}) }}"
-                     style="background-color: #000000;">
-                    <i class="play-icon"></i>
-                </div>
-            </a>
+            {% if block.media.isVideo() %}
```
Comment (HeahDude) [inline templates/home/articles.html.twig]: Same as above.

File: templates/home/articles.html.twig [hunk 8]
```diff
@@ -1,14 +1,21 @@
...
+                <video controls width="300" height="200" />
+                    <source src="{{ media_asset(block.media) }}" type="{{ block.media.mimeType }}">
+                    Votre navigateur ne prend pas en charge la balise "vidéo".
```
Comment (HeahDude) [inline templates/home/articles.html.twig]: Please remove the accent for the tag name.

File: templates/home/articles.html.twig [hunk 9]
```diff
@@ -50,13 +57,20 @@
...
     {% else %}
         <div class="cs card--small {{ block.hasDisplayFilter ? '' : ' card--no-gradient' }}">
-            <a href="{{ block.link }}" class="background-container">
-                <div class="background card__background progressive-background"
-                     data-sd="{{ media_asset(block.media, {'w': 300, 'q': 20}) }}"
-                     data-hd="{{ media_asset(block.media, {'w': 1200, 'q': 70}) }}"
-                     style="background-color: #000000;">
-                </div>
-            </a>
+            {% if block.media.isVideo() %}
```
Comment (HeahDude) [inline templates/home/articles.html.twig]: Same comment as before.

**Pull request merged**